File: BusReservations/BusReservations.WebAPI/DTOs/ReservationGetDto.cs
using BusReservations.Core.Domain;
using BusReservations.Core.Domain.SeatModel;

namespace BusReservations.WebAPI.DTOs
{
    public class ReservationGetDto
    {
        public UserGetDto User { get; set; }

        public DrivenRouteSimpleDto DrivenRoute { get; set; }
        public BusSimpleDto Bus { get; set; }
        public SeatDto Seat { get; set; }
        public float FinalSeatPrice { get; set; }
    }
}

File: Frontend/BusReservationsFrontend/src/app/models/reservationGetDto.ts
import { Bus } from './bus';
import { BusDrivenRoute } from './busDrivenRoute';
import { DrivenRoute } from './drivenRoute';
import { Seat } from './seat';
import { User } from './user';

export interface ReservationGetDto {
  id: string;
  user: User;
  busDrivenRoute: BusDrivenRoute;
  seat: Seat;
  finalSeatPrice: number;
}
